lightning-persister/src/lib.rs

```diff
@@ -53,12 +52,12 @@ impl<Signer: Sign> DiskWriteable for ChannelMonitor<Signer> {
 	}
 }
 
-impl<Signer: Sign, M, T, K, F, L> DiskWriteable for ChannelManager<Signer, Arc<M>, Arc<T>, Arc<K>, Arc<F>, Arc<L>>
-where M: chain::Watch<Signer>,
-      T: BroadcasterInterface,
-      K: KeysInterface<Signer=Signer>,
-      F: FeeEstimator,
-      L: Logger,
+impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> DiskWriteable for ChannelManager<Signer, M, T, K, F, L>
+where M::Target: chain::Watch<Signer>,
+      T::Target: BroadcasterInterface,
+      K::Target: KeysInterface<Signer=Signer>,
+      F::Target: FeeEstimator,
+      L::Target: Logger
```

Comment:
nit: could you format according to these guidelines?

https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/items.md#where-clauses

i.e., move the first item in the where clause to its own line, ident only with tabs, and use a trailing comma for all items (including the last one).

I know we haven't been very consistent, but this feels like a good convention to follow given it sticks to using only tabs and seems to make editors happy.